.github/workflows/CI.yml

```diff
@@ -7,6 +8,13 @@ on:
     branches:
       - master
     tags: '*'
+
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```

Comment:
This will cancel running tests if a new commit is pushed to a PR. Particularly useful for the longer running integration tests but also helpful to reduce the load on Github actions for our regular tests.